src/components/menu-bar/menu-bar.jsx

```diff
@@ -641,7 +675,11 @@ const mapDispatchToProps = dispatch => ({
     onRequestCloseLanguage: () => dispatch(closeLanguageMenu()),
     onClickLogin: () => dispatch(openLoginMenu()),
     onRequestCloseLogin: () => dispatch(closeLoginMenu()),
-    onSeeCommunity: () => dispatch(setPlayer(true))
+    onSeeCommunity: () => dispatch(setPlayer(true)),
+    onShare: () => {}, // NOTE: implement this
+    startSaving: () => dispatch(startSaving()),
+    stepTowardsCreatingNewProject: () => dispatch(stepTowardsCreatingNewProject()),
+    stepTowardsNewProjectWithoutSaving: () => dispatch(stepTowardsNewProjectWithoutSaving())
```

Comment:
what those names miss for me is the clarity about whether they are at the beginning of the action or the end. Does "onSave" happen when a save has happened? or when a save is going to happen? "onClickSaveNow" is clear on that point --- but then it can't be also used for when we want to initiate a save in other situations, like when we are planning to replace the project in the vm with another one.

Reply:
In this case, I think I could have menu-bar pass a flag to the reducer like

onClickNew(shouldSave)